src/typeobject.js

```diff
@@ -0,0 +1,148 @@
+Sk.builtin.type_is_subtype_base_chain = function type_is_subtype_base_chain(a, b) {
+    do {
+        if (a == b)
+            return true;
+        a = a.tp$base;
+    } while (!a === undefined);
+
+    return (b == Sk.builtin.object);
+};
+
+Sk.builtin.PyType_IsSubtype = function PyType_IsSubtype(a, b) {
+    var mro = a.tp$mro;
+    if (mro) {
+        /* Deal with multiple inheritance without recursion
+           by walking the MRO tuple */
+        goog.asserts.assert(mro instanceof Sk.builtin.tuple);
+        for (var i = 0; i < mro.v.length; i++) {
+            if (mro.v[i] == b) {
+                return true; 
+            }
+        }
+        return false; 
+    }
+    else {
+        /* a is not completely initilized yet; follow tp_base */
+        return Sk.builtin.type_is_subtype_base_chain(a, b);
+    }
+}
+
+/**
+ * @constructor
+ * Sk.builtin.superbi
+ */
+Sk.builtin.superbi = function superbi (a_type, self) {
```

Comment:
Like I said, I wasn't blaming you for this, I know it was already called ```superbi```.  I was just commenting that maybe we should fix it now since you're fixing it up anyway.  This is not a deal breaker for me, but it does seem like it would be nice to move things towards consistency.  Your call, as far as I'm concerned.